src/st/dre/14774.c

```diff
@@ -137,7 +137,140 @@ void func_801961DC(s16 arg0) {
     }
 }
 
-INCLUDE_ASM("asm/us/st/dre/nonmatchings/14774", EntityUnkId21);
+extern s32 D_80097400;
+
+void EntityCSMoveAlucard(Entity* self) {
+    Unkstruct8* currentRoomTileLayout = &g_CurrentRoomTileLayout;
+    Entity* player = &PLAYER;
+    s16 posX;
+
+    FntPrint("step %x\n", self->step);
+    if ((D_801A3ED4 != 0) && (self->step < 8)) {
+        SetStep(8);
+    }
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_8018047C);
+        D_8003C8B8 = 0;
+        D_80097400 = 1;
+        g_Player.D_80072EF4 = 0;
+        g_Player.D_80072EFC = 1;
+        if (g_DemoMode != Demo_None) {
+            self->ext.generic.unk7C.s = 64;
+            break;
+        }
+        self->ext.generic.unk7C.s = 128;
+        break;
+
+    case 1:
+        g_Player.D_80072EF4 = 0;
+        if (self->ext.generic.unk7C.u != 0) {
+            self->ext.generic.unk7C.u--;
+        }
+        if (g_Player.unk0C & 7) {
```

Comment:
I tried a couple of simplifications here but they didn't match. This seems to be the scratch https://decomp.me/scratch/T8Auk